addons/l10n_it_edi/models/ir_attachment.py

```diff
@@ -1,20 +1,115 @@
 # -*- coding: utf-8 -*-
 
-from odoo import api, models
+from odoo import _, api, models, tools
 from odoo.addons.l10n_it_edi.tools.remove_signature import remove_signature
+from odoo.exceptions import UserError
 
 from lxml import etree
+from io import BytesIO
+import base64
+import binascii
 import logging
 import re
 
 _logger = logging.getLogger(__name__)
 
+try:
+    from asn1crypto import cms
+except (OSError, ImportError) as err:
+    _logger.debug(err)
+
 FATTURAPA_FILENAME_RE = "[A-Z]{2}[A-Za-z0-9]{2,28}_[A-Za-z0-9]{0,5}.((?i:xml.p7m|xml))"
+re_base64 = re.compile(rb"^([A-Za-z0-9+/]{4})*([A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{2}==)?$")
+
+
+def is_base64(s):
+    s = s or b""
+    s = s.replace(b"\r", b"").replace(b"\n", b"")
+    return re_base64.match(s)
 
 
 class IrAttachment(models.Model):
     _inherit = 'ir.attachment'
 
+    @api.model
+    def remove_xades_sign(self, xml):
+        # Recovering parser is needed for files where strings like
+        # xmlns:ds="http://www.w3.org/2000/09/xmldsig#&quot;"
+        # are present: even if lxml raises
+        # {XMLSyntaxError}xmlns:ds:
+        # 'http://www.w3.org/2000/09/xmldsig#"' is not a valid URI
+        # such files are accepted by SDI
+        recovering_parser = etree.XMLParser(recover=True, resolve_entities=False)
+        root = etree.XML(xml, parser=recovering_parser)
+        for elem in root.iter("*"):
+            if elem.tag.find("Signature") > -1:
+                elem.getparent().remove(elem)
+                break
+            if any(" " in elem.nsmap[tag] for tag in elem.nsmap):
+                etree.cleanup_namespaces(elem)
+        return etree.tostring(root)
+
+    @api.model
+    def strip_xml_content(self, xml):
+        recovering_parser = etree.XMLParser(recover=True, resolve_entities=False)
+        root = etree.XML(xml, parser=recovering_parser)
+        return etree.tostring(root)
+
+    @staticmethod
+    def extract_cades(data):
+        info = cms.ContentInfo.load(data)
+        return info["content"]["encap_content_info"]["content"].native
+
+    def cleanup_xml(self, xml_string):
+        xml_string = self.remove_xades_sign(xml_string)
+        xml_string = self.strip_xml_content(xml_string)
+        return xml_string
+
+    def get_xml_string(self):
+        try:
+            data = base64.b64decode(self.datas)
+        except binascii.Error as e:
+            raise UserError(_("Corrupted attachment %s.") % e.args) from e
```

Comment:
```suggestion
        data = self.raw
```

![image](https://github.com/odoo/odoo/assets/1665365/0201d23d-cb3e-433a-99d2-af6f8e034fb3)